### PR TITLE
Bugfix/platfowner/bugfix

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Add state versions to node status report to tracker server
- Improve and rename createRandomVersion() -> createUniqueVersionName()
- Destroy temp version for segment merge
- Rename temp versions
- Refactor addNewBlockToChain() by separating tx execution from block addition

#147

### CODE REVIEW

File: blockchain/index.js [hunk 1]
```diff
@@ -259,46 +249,40 @@ class Blockchain {
     return chainSegment.length > 0 ? chainSegment : [];
   }
 
-  merge(chainSegment, db) {
+  getValidBlocks(chainSegment) {
```
Comment (minsulee2) [inline blockchain/index.js:252]: Wow, I love it !!

File: db/state-manager.js [hunk 2]
```diff
@@ -227,8 +227,15 @@ class StateManager {
    * Returns a random state version with the given version prefix.
    * @param {string} versionPrefix version prefix
    */
-  static createRandomVersion(versionPrefix) {
-    return `${versionPrefix}:${Date.now()}:${Math.floor(Math.random() * 10000)}`;
+  createRandomVersion(versionPrefix) {
```
Comment (minsulee2) [inline db/state-manager.js:230]: What about createVersionWithIdx or something? It doesn't seem random value version anymore..

**Review by minsulee2 — APPROVED** [review]
LGTM with minor comments!

File: db/state-manager.js (hunk 2 above)
Comment (platfowner) [reply db/state-manager.js]: Tried to pick createUniqueVersionName(). Thanks.

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Comment by platfowner** [comment]
Thanks for fast review!

**Pull request merged**